compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -9109,8 +9113,8 @@ private STNode parseClientResourceAccessOrAsyncSendActionRhs(STNode expression,
                     // This logic is added to improve recovery for resource method call action slash token
                     // Next token is a Missing token means, it is a correct token recovered previously
                     SyntaxKind nextNextNextTokenKind = getNextNextToken().kind;
-                    if (nextNextNextTokenKind == SyntaxKind.OPEN_PAREN_TOKEN || 
-                            isEndOfActionOrExpression(nextNextNextTokenKind, isRhsExpr, isInMatchGuard) || 
+                    if (nextNextNextTokenKind == SyntaxKind.OPEN_PAREN_TOKEN ||
+                            isEndOfActionOrExpression(getNextNextToken(), isRhsExpr, isInMatchGuard) ||
```

Comment:
We are calling `getNextNextToken()` twice here and L9111. 
```suggestion
                    if (nextNextToken.kind == SyntaxKind.OPEN_PAREN_TOKEN ||
                            isEndOfActionOrExpression(nextNextToken, isRhsExpr, isInMatchGuard) ||
```